### ISSUE TITLE
CLI `--lr_scheduler` doesn't work?

### ISSUE THREAD

**Issue by profPlum** [Lightning-AI/pytorch-lightning#18663]
### Bug description

When using LightningCLI + `--lr_scheduler pytorch_lightning.cli.ReduceLROnPlateau --lr_scheduler.monitor=epoch` : learning rate does not change during training (even when progress is stalled). I've tested it using the LR monitor on 10 different experiments and LR does not change.

### What version are you seeing the problem on?

v1.9

### How to reproduce the bug
Use LightningCLI + `--lr_scheduler pytorch_lightning.cli.ReduceLROnPlateau --lr_scheduler.monitor=epoch` + 
```python
def configure_optimizers(self):
   return torch.optim.Adam(self.parameters(), lr=self.learning_rate)
```

### Error messages and logs

<img width="376" alt="image" src="https://github.com/Lightning-AI/lightning/assets/4739054/27e43a34-f53c-4e16-81fe-04b2a080eaf2">
<img width="853" alt="image" src="https://github.com/Lightning-AI/lightning/assets/4739054/4875ce53-eb50-411c-8509-748f101f9ed6">

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component: LightningModule or Trainer?
#- PyTorch Lightning Version (e.g., 1.5.0): 1.9.0
#- PyTorch Version (e.g., 2.0): 1.10.0
#- Python version (e.g., 3.9): 3.8.16
#- OS (e.g., Linux): Linux
#- GPU models and configuration: V100
#- How you installed Lightning(`conda`, `pip`, source): pip
```

</details>

### More info

I've seen it happen in version 1.9.0 but it's quite possible it applies to master as well.

cc @borda @carmocca @mauvilsa

**Comment by profPlum** [comment]
UPDATE: I've just tried `import pytorch_lightning.cli.ReduceLROnPlateau`
and I get an error that pytorch_lightning.cli doesn't exist.

**Despite this** the LightningCLI --help docs say that pytorch_lightning.cli.ReduceLROnPlateau is one of the valid classes to pass to the  --lr_scheduler argument, also it didn't terminate on error when passing this class (which apparently might not be accessible?) to the CLI arg mentioned.

<img width="696" alt="image" src="https://github.com/Lightning-AI/lightning/assets/4739054/55a86139-eded-411e-92de-a715c423fd12">

**Comment by awaelchli** [comment]
@profPlum Is this possibly a duplicate of #17800? Could you check the responses there?

**Comment by profPlum** [comment]
@awaelchli It seems it is. However I'd like to call attention to the fact that the docs [weren't fully updated](https://lightning.ai/docs/pytorch/LTS/cli/lightning_cli_intermediate_2.html?highlight=pl%20cli%20reducelronplateau). The link I gave shows an example with `python main.py fit --lr_scheduler=ReduceLROnPlateau --lr_scheduler.monitor=epoch`

Here `--optimizer` is not used & I'm pretty sure it's an error to monitor the epoch? I initially had copied this code into my CLI call verbatim but then I realized that I'm pretty sure the monitor argument is the metric to monitor for a plateau? If that's so then epoch will obviously never plateau which makes this example misleading in 2 ways (unless I'm mistaken).

**Comment by profPlum** [comment]
@awaelchli wait to clarify regarding the other issue: does this mean that [LR finder](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.LearningRateFinder.html#learningratefinder) isn't supported when supplying optimizer and/or LR scheduler on the CLI?

The docs for LR finder mention that you need to take lr/learning_rate as a LightningModule __init__ arg & set them as an attribute to the LightningModule in order for it to work.

**Comment by carmocca** [comment]
@profPlum Those are all valid observations. Feel free to open PRs updating/fixing them!
